src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -1534,48 +1565,18 @@ public SolutionCompilationState RemoveAnalyzerConfigDocuments(ImmutableArray<Doc
 
         // The documents might be contributing to multiple different projects; split them by project and then we'll
         // process one project at a time.
-        return UpdateDocumentsInMultipleProjects(
+        return WithDocumentStatesOfMultipleProjects(
             documentInfos.GroupBy(d => d.Id.ProjectId).Select(g =>
             {
                 var projectId = g.Key;
-                this.SolutionState.CheckContainsProject(projectId);
-                var projectState = this.SolutionState.GetRequiredProjectState(projectId);
-                return (projectId, newDocumentStates: g.SelectAsArray(di => createDocumentState(di, projectState)));
+                SolutionState.CheckContainsProject(projectId);
+                var projectState = SolutionState.GetRequiredProjectState(projectId);
+                return (projectId, newDocumentStates: g.SelectAsArray(projectState.CreateDocument<TDocumentState>));
```

Comment:
>SelectAsArray [](http://example.com/codeflow?start=56&length=13)

nit: would be nice if this used the arg version of SelectAsArray

Reply:
We actually do not have TArg version of IEnumerable.SelectAsArray. If needed let's add it in another PR.

Reply:
oh, is that only available at the compiler level? I'm a bit surprised if so.

Reply:
Yep, you're right that we don't have an IEnumerable arg version.